generated-code-samples-for-review/speech/speech_transcribe_async.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Code generated by protoc-gen-go_gapic. DO NOT EDIT.
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+
+	speech "cloud.google.com/go/speech/apiv1"
+	speechpb "google.golang.org/genproto/googleapis/cloud/speech/v1"
+)
+
+// [START speech_transcribe_async]
+
+// sampleLongRunningRecognize: Transcribe a long audio file using asynchronous speech recognition
+//
+// localFilePath: Path to local audio file, e.g. /path/audio.wav
+func sampleLongRunningRecognize(localFilePath string) error {
+	ctx := context.Background()
+	c, err := speech.NewClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	// localFilePath := "resources/brooklyn_bridge.raw"
+	// Path to local audio file, e.g. /path/audio.wav
+	localFilePathBytes, err := ioutil.ReadFile(localFilePath)
+	if err != nil {
+		return err
+	}
+
+	req := &speechpb.LongRunningRecognizeRequest{
+		Audio: &speechpb.RecognitionAudio{
+			AudioSource: &speechpb.RecognitionAudio_Content{
+				Content: localFilePathBytes,
+			},
+		},
+		Config: &speechpb.RecognitionConfig{
+			// The language of the supplied audio
+			LanguageCode: "en-US",
+			// Sample rate in Hertz of the audio data sent
+			SampleRateHertz: 16000,
+			// Encoding of audio data sent. This sample sets this explicitly.
+			// This field is optional for FLAC and WAV audio formats.
+			Encoding: speechpb.RecognitionConfig_LINEAR16,
+		},
+	}
+	op, err := c.LongRunningRecognize(ctx, req)
+	if err != nil {
+		return err
+	}
+
+	resp, err := op.Wait(ctx)
+	if err != nil {
+		return err
+	}
+
+	for _, result := range resp.GetResults() {
+		// First alternative is the most probable result
+		alternative := result.GetAlternatives()[0]
```

Comment:
Is it possible the list of alternatives would be empty? If so, this would panic.

Reply:
This is how the sample is configured.

Reply:
Yeah, we'd only want to have this in samples where we're pretty sure there should be alternatives. Looking to Go developers to be able to debug when it blows up, otherwise I believe every single array access would need error handling (_presumably good and standard for production, presumably less good for embedded website code_)

The [response configuration](https://github.com/googleapis/googleapis/blob/master/google/cloud/speech/v1/samples/speech_transcribe_async.yaml) FYI

```yaml
  response:
  - loop:
      variable: result
      collection: $resp.results
      body:
      - comment:
        - First alternative is the most probable result
      - define: alternative = result.alternatives[0]
      - print:
        - "Transcript: %s"
        - alternative.transcript
```